src/client/handlers/VaultsSecretsList.ts

```diff
@@ -2,55 +2,65 @@ import type { DB } from '@matrixai/db';
 import type {
   ClientRPCRequestParams,
   ClientRPCResponseResult,
-  SecretNameMessage,
-  VaultIdentifierMessage,
+  SecretFilesMessage,
+  SecretFiles,
 } from '../types';
 import type VaultManager from '../../vaults/VaultManager';
+import path from 'path';
 import { ServerHandler } from '@matrixai/rpc';
 import * as vaultsUtils from '../../vaults/utils';
 import * as vaultsErrors from '../../vaults/errors';
-import * as vaultOps from '../../vaults/VaultOps';
+import * as clientErrors from '../errors';
 
 class VaultsSecretsList extends ServerHandler<
   {
     vaultManager: VaultManager;
     db: DB;
   },
-  ClientRPCRequestParams<VaultIdentifierMessage>,
-  ClientRPCResponseResult<SecretNameMessage>
+  ClientRPCRequestParams<SecretFilesMessage>,
+  ClientRPCResponseResult<SecretFiles>
 > {
   public async *handle(
-    input: ClientRPCRequestParams<VaultIdentifierMessage>,
-    _cancel,
-    _meta,
-    ctx,
-  ): AsyncGenerator<ClientRPCResponseResult<SecretNameMessage>> {
-    if (ctx.signal.aborted) throw ctx.signal.reason;
+    input: ClientRPCRequestParams<SecretFilesMessage>,
+    _cancel: any,
+  ): AsyncGenerator<ClientRPCResponseResult<SecretFiles>, void, void> {
     const { vaultManager, db } = this.container;
-    const secrets = await db.withTransactionF(async (tran) => {
+    const vaultId = await db.withTransactionF(async (tran) => {
       const vaultIdFromName = await vaultManager.getVaultId(
         input.nameOrId,
         tran,
       );
       const vaultId =
         vaultIdFromName ?? vaultsUtils.decodeVaultId(input.nameOrId);
-      if (vaultId == null) {
-        throw new vaultsErrors.ErrorVaultsVaultUndefined();
-      }
-      return await vaultManager.withVaults(
-        [vaultId],
-        async (vault) => {
-          return await vaultOps.listSecrets(vault);
-        },
-        tran,
-      );
+      if (vaultId == null) throw new vaultsErrors.ErrorVaultsVaultUndefined();
+      return vaultId;
+    });
+
+    yield* vaultManager.withVaultsG([vaultId], (vault) => {
+      return vault.readG(async function* (fs): AsyncGenerator<
+        SecretFiles,
+        void,
+        void
+      > {
+        let files: Array<string>;
+        try {
+          // @ts-ignore: While the types don't fully match, it matches enough for our usage.
+          files = await fs.promises.readdir(input.path);
+        } catch (e) {
+          throw new clientErrors.ErrorClientFSReadFailed(
+            `No matches exist for path: ${input.path}`,
+            { cause: e },
+          );
+        }
```

Comment:
Two points.

1. Avoid using the `@ts-ignore` here. Needing a `@ts-ignore` is a last ditch solution. In this case we can take both the outputs of `string` and `Buffer` and convert it back to the `string` format with `.toString()`.
2. I don't think we should wrap the error here? In any case never assume that it will be the error you expect. If you want to wrap an `ENOENT` error then you need to check it's that error. If it isn't then just re-throw the unchanged error.

Reply:
> I don't think we should wrap the error here? In any case never assume that it will be the error you expect. If you want to wrap an ENOENT error then you need to check it's that error. If it isn't then just re-throw the unchanged error.

Okay, I'll just remove the error wrapping here then.

Reply:
Apparently this error isnt registered to be thrown as an error in Polykey, so while the RPC will throw, the handler will not interpret it as an error, rather just a `Object`. As such, I will be wrapping the error around something like `vaultErrors.ErrorVaultSecretUndefined`.